src/python_framework_v02/troute/nhd_io.py

```diff
@@ -1084,3 +1086,73 @@ def build_coastal_ncdf_dataframe(coastal_ncdf):
     with xr.open_dataset(coastal_ncdf) as ds:
         coastal_ncdf_df = ds[["elev", "depth"]]
         return coastal_ncdf_df.to_dataframe()
+
+def lastobs_df_output(dt,
+    nts,
+    run_results,
+    wrf_hydro_lastobs_file,
+    q0,
+    gages,
+    run_set_iterator,
+    lastobs_output_title,
+    lastobs_output_folder=False,
+    ):
+    lastobs_df = new_lastobs(run_results, dt * nts)
+    lastobs_df.index.names = ['link']
+    lastobs_df.insert(0, 'last_model_discharge', q0.loc[lastobs_df.index]['qu0'])
+    lastobs_df.insert(0, 'gages', gages)
+    if lastobs_output_folder:
+        lastobs_string = lastobs_output_folder+lastobs_output_title
+        # lastobs_string = lastobs_output_folder+"lastobs_df_"+str(run_set_iterator)+".nc"
+    else:
+        lastobs_string = lastobs_output_title
+        # lastobs_string = "lastobs_df_"+str(run_set_iterator)+".nc"
+    if not 'Nudge' in lastobs_df.columns:
+        lastobs_df.insert(loc=0, column='Nudge', value=np.NaN)
+    lastobs_df['Nudge'] = np.NaN
+    lastobs_df = lastobs_df.to_xarray()
+    lastobs_df.attrs['modelTimeAtOutput'] = lastobs_output_title[:-19]
+    lastobs_df.to_netcdf(lastobs_string)
```

Comment:
Can you please re-open one of the netcdf files generated here as an xarray dataset and print the contents of the xarray object in a comment on this PR. Then, do the same thing with an already-existing lastobs file taken off the shelf. It would be useful to compare the dimensions, variables and attributes across files.